### PR TITLE
feat(etherscan): Add get_block_by_timestamp to ethers-etherscan

### PR DESCRIPTION
**Pull request by 0xrainbowtrout**
## Motivation

The Etherscan API has a blocks endpoint with a variety of useful methods. This PR incorporates the `get-block-number-by-timestamp` [endpoint](https://docs.etherscan.io/api-endpoints/blocks#get-block-number-by-timestamp). I need to be able to fetch blocks by timestamps to evaluate blockchain state by timestamp and this seems like the best home for that functionality given the `ethers-etherscan` module.

## Solution

Add the API endpoint.

## PR Checklist

-   [x] Added Tests
-   [x] Added Documentation
-   ~[ ] Breaking changes~ (No breaking changes)

### CODE REVIEW

**Comment by 0xrainbowtrout** [comment]
Just a ping here -- No rush but didn't want it to fall off if you hadn't seen it.

File: ethers-etherscan/src/blocks.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+use std::collections::HashMap;
+use std::str::FromStr;
+
```
Comment (mattsse) [inline ethers-etherscan/src/blocks.rs:3]:
```suggestion
```

File: ethers-etherscan/src/blocks.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+use ethers_core::types::BlockNumber;
+use serde::{Deserialize, Serialize};
+
```
Comment (mattsse) [inline ethers-etherscan/src/blocks.rs:6]:
```suggestion
```

**Review by mattsse — APPROVED** [review]
lgtm

**Comment by 0xrainbowtrout** [comment]
Fixed formatting from the failed clippy/fmt workflows.

**Review by DaniPopes — APPROVED** [review]
Thanks!

**Pull request merged**